lib/matplotlib/cbook/__init__.py

```diff
@@ -302,11 +302,17 @@ def local_over_kwdict(local_var, kwargs, *keys):
 
 
 def strip_math(s):
-    """remove latex formatting from mathtext"""
-    remove = (r'\mathdefault', r'\rm', r'\cal', r'\tt', r'\it', '\\', '{', '}')
-    s = s[1:-1]
-    for r in remove:
-        s = s.replace(r, '')
+    """
+    Remove latex formatting from mathtext.
+
+    Only handles fully math and fully non-math strings.
+    """
+    if s[:1] == s[-1:] == "$":
```

Comment:
Any reason not to write `if s[0] == s[-1] == "$"`

Reply:
That was to handle empty strings as well (which do occur here if e.g. you disable offsettexts), but I realized this would still give the wrong result with s = "$".  So I added an explicit check on len(s) instead.